conda_build/config.py

```diff
@@ -913,8 +913,7 @@ def __exit__(self, e_type, e_value, traceback):
             self.clean(remove_folders=False)
 
 
-def get_or_merge_config(config, variant=None, **kwargs):
-    """Always returns a new object - never changes the config that might be passed in."""
+def _get_or_merge_config(config, variant=None, **kwargs):
```

Comment:
I'm confused about this. Why do we need the private indirection? Just to monkey patch it? If that's the case, it might be good to drop a comment here so future folks are not tempted to undo it :D

Reply:
Yes, good idea, thanks.